resource/js/copy-to-clipboard.js

```diff
@@ -0,0 +1,30 @@
+// function for copying the content from a specific element (by id) to the clipboard
+function copyToClipboard (id) {
+  const copyElem = document.getElementById(id)
+  const sel = window.getSelection()
+  const range = document.createRange()
+  range.selectNodeContents(copyElem)
+  sel.removeAllRanges()
+  sel.addRange(range)
+
+  navigator.clipboard.writeText(copyElem.innerText).catch((err) =>
+    console.error('Failed to copy text to clipboard: ', err))
+}
+
+function registerCopyToClipboardEvents () {
+  const copyPrefElem = document.getElementById('copy-preflabel')
+  if (copyPrefElem) {
+    copyPrefElem.addEventListener('click', () => copyToClipboard('concept-preflabel'))
+  }
+
+  const copyUriElem = document.getElementById('copy-uri')
+  if (copyUriElem) {
+    copyUriElem.addEventListener('click', () => copyToClipboard('concept-uri'))
+  }
+}
+
+// register the copyToClipboard function as event an handler for the copy buttons
+registerCopyToClipboardEvents()
+
+// re-register the event handlers after partial page loads
```

Comment:
This line fills me with dread. We should make sure all event handlers are re-registered collectively. Partial page loading as a specific test case hasn't been used as of yet. Should we have an issue about it?

Reply:
I agree that this is a more general issue, but currently we don't have that many event handlers and AFAICT only the concept-mappings component needs to re-register its event handler after a partial page load, and it already does that.